src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -799,13 +799,46 @@ AdjustDistributedExecutionAfterLocalExecution(DistributedExecution *execution)
 
 /*
  * ExecuteUtilityTaskListWithoutResults is a wrapper around executing task
- * list for utility commands. It simply calls in adaptive executor's task
- * execution function.
+ * list for utility commands. For remote tasks, it simply calls in adaptive
+ * executor's task execution function. For local tasks (if any), kicks Process
+ * Utility via CitusProcessUtility for utility commands. As some local utility
+ * commands can trigger udf calls, this function also processes those udf calls
+ * locally.
  */
 void
-ExecuteUtilityTaskListWithoutResults(List *taskList)
+ExecuteUtilityTaskListWithoutResults(List *taskList, bool localExecutionSupported)
 {
-	ExecuteTaskList(ROW_MODIFY_NONE, taskList, MaxAdaptiveExecutorPoolSize);
+	RowModifyLevel rowModifyLevel = ROW_MODIFY_NONE;
+
+	List *localTaskList = NIL;
+	List *remoteTaskList = NIL;
+
+	bool readOnlyPlan = !TaskListModifiesDatabase(rowModifyLevel, taskList);
+
+	/* divide tasks into two if localExecutionSupported is set to true */
+	if (localExecutionSupported)
+	{
+		/* set local (if any) & remote tasks */
+		ExtractLocalAndRemoteTasks(readOnlyPlan, taskList, &localTaskList,
+								   &remoteTaskList);
+	}
+
+	if (ShouldExecuteTasksLocally(localTaskList))
```

Comment:
I think `ExtractLocalAndRemoteTasks ` should happen only when `ShouldExecuteTasksLocally ()` returns true. Otherwise, all tasks should be remote